### PR TITLE
Add "local" provider for http-challenge

### PR DESCRIPTION
**Pull request by avalor1**
Adds a local provider for http-challenge to allow http validation via a webserver
- split http-challenge into s3 and local playbooks which get included when letsencrypt_http_provider is set accordingly
- add letsencrypt_local_validation_path variable to defaults to work out of the box
- update documentation with new local provider for http-challenge
- update examples with letsencrypt_http_provider variable

### CODE REVIEW

File: roles/letsencrypt/README.md [hunk 1]
```diff
@@ -14,7 +14,13 @@ ansible-galaxy collection install t_systems_mms.letsencrypt
 
 ## http-challenge
 
-It currently uses an AWS S3 bucket to safe the hashfiles. You have to set a redirect rule in your proxy or webserver to allow the acme challenge bot to read the file, during the http-01 challenge to work:
+The validation via http-challenge currently supports the usage of a local path at the webserver and also an AWS S3 bucket to safe the hashfiles
```
Comment (rndmh3ro) [inline roles/letsencrypt/README.md:17]:
```suggestion
The validation via http-challenge currently supports the usage of a local path at the webserver and also an AWS S3 bucket to safe the hashfiles.
```

File: roles/letsencrypt/README.md [hunk 2]
```diff
@@ -14,7 +14,13 @@ ansible-galaxy collection install t_systems_mms.letsencrypt
...
+
+### local path
+When using a local path at the webserver you have to ensure that this file is reachable by your configured vhost(s)
```
Comment (rndmh3ro) [inline roles/letsencrypt/README.md:20]:
```suggestion
When using a local path at the webserver you have to ensure that this file is reachable by your configured vhost(s).
```

**Review by rndmh3ro — COMMENTED** [review]
Should we somehow start splitting the docs for the different challenge types?

**Comment by avalor1** [comment]
> Should we somehow start splitting the docs for the different challenge types?

Yeah, you are right. It seems to be a bit cluttered right now. Will split all into their own readme with one Main README which just contains the links.

File: roles/letsencrypt/README.md [hunk 3]
```diff
@@ -14,50 +14,24 @@ ansible-galaxy collection install t_systems_mms.letsencrypt
...
-rewrite (\.well-known/acme-challenge.*) https://letsencrypt-challenge-bucket.s3.amazonaws.com/$1
-```
+Please see [the corresponding README](roles/letsencrypt/README-http-challenge.md) for usage hints and variables of the HTTP-01 challenge
 
 ## dns-challenge
+
 Currently the role supports the InternetX autodns API and the Azure DNS API. Feel free to contribute with other DNS APIs.
 
-## Variables for DNS & HTTP challenge
+Please see [the corresponding README](roles/letsencrypt/README-dns-challenge.md) for variables of the DNS-01 challenge
```
Comment (rndmh3ro) [inline roles/letsencrypt/README.md:25]: that's a relative path. Try clicking it here: https://github.com/T-Systems-MMS/ansible-collection-letsencrypt/blob/provider_for_local_path/roles/letsencrypt/README.md

File: roles/letsencrypt/README.md [hunk 4]
```diff
@@ -14,50 +14,24 @@ ansible-galaxy collection install t_systems_mms.letsencrypt
...
-**Nginx:**
-
-```bash
-rewrite (\.well-known/acme-challenge.*) https://letsencrypt-challenge-bucket.s3.amazonaws.com/$1
-```
+Please see [the corresponding README](roles/letsencrypt/README-http-challenge.md) for usage hints and variables of the HTTP-01 challenge
 
 ## dns-challenge
+
 Currently the role supports the InternetX autodns API and the Azure DNS API. Feel free to contribute with other DNS APIs.
```
Comment (rndmh3ro) [inline roles/letsencrypt/README.md:23]: we also support hetzner: https://github.com/T-Systems-MMS/ansible-collection-letsencrypt/pull/22

File: roles/letsencrypt/tasks/http-challenge-local.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (rndmh3ro) [inline roles/letsencrypt/tasks/http-challenge-local.yml:1]: We can test this challenge in the CI, right?

**Review by rndmh3ro — CHANGES_REQUESTED** [review]

File: roles/letsencrypt/README.md (hunk 3 above)
Comment (avalor1) [reply roles/letsencrypt/README.md:25]: yep, missed that

File: roles/letsencrypt/README.md (hunk 4 above)
Comment (avalor1) [reply roles/letsencrypt/README.md]: addet it to the description

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -94,6 +94,15 @@ jobs:
       - name: Set up Pebble
         run: curl --request POST --data '{"ip":"10.30.50.1"}' http://localhost:8055/set-default-ipv4
 
+      - name: update apt package list
+        run: apt update
+
+      - name: install apache2
```
Comment (rndmh3ro) [inline .github/workflows/main.yml:100]: This should probably happen in a docker-container..
Start nginx or apache with a custom configuration and mount/copy the hash-file into the container.

Comment (avalor1) [reply .github/workflows/main.yml:100]: i hoped it would work this way. very naive :D

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -82,6 +82,14 @@ jobs:
           - 2.7
           - 3.7
           - 3.8
+    # start nginx as service to test validation via http-challenge local path
+    services:
+      nginx:
+        image: nginx
+        ports:
+          - 5002:80
+        options: -v /var/tmp:/usr/share/nginx/html
```
Comment (rndmh3ro) [inline .github/workflows/main.yml:91]: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservicesservice_idvolumes

File: roles/letsencrypt/README.md [hunk 8]
```diff
@@ -102,7 +101,7 @@ wildcard-certificates use base name again as otherwise DNS txt record creation c
 | private_key_path                         | yes      | $letsencrypt_conf_dir/certs          | path for private key
 | remaining_days                           | yes      | 30                                   | min days remaining before certificate will be renewed
 | convert_cert_to                          | no       |                                      | format to convert the certificate to: `pfx`
-
+| acme_directory_validate_certs            | no       | true                                 | Only exists to enable integration tests of http challenges with pebble server
```
Comment (rndmh3ro) [inline roles/letsencrypt/README.md:104]: We already have a variable for this here:
https://github.com/T-Systems-MMS/ansible-collection-letsencrypt/blob/master/roles/letsencrypt/tasks/dns-challenge-pebble.yml#L40
https://github.com/T-Systems-MMS/ansible-collection-letsencrypt/blob/master/tests/integration/targets/letsencrypt/test.yml#L20

Comment (avalor1) [reply roles/letsencrypt/README.md]: fixed with 41f3150

File: roles/letsencrypt/tasks/http-challenge-local.yml (hunk 5 above)
Comment (avalor1) [reply roles/letsencrypt/tasks/http-challenge-local.yml]: yep, works now :) 3a7dbbb

File: .github/workflows/main.yml (hunk 6 above)
Comment (avalor1) [reply .github/workflows/main.yml]: using a service now 66265e6

File: .github/workflows/main.yml (hunk 7 above)
Comment (avalor1) [reply .github/workflows/main.yml]: reverted back to "volumes" as this parameter works as intended b87bbbb

**Review by rndmh3ro — APPROVED** [review]

**Pull request merged**